packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -141,17 +189,45 @@ class OnsetPicker extends React.Component {
   changeGranularity(granularity) {
     if (granularity === GranularityOptions.AGE) { // Calculate age values and update onsetDate to match age calculation
       this.setState((prevState) => {
-        const ageValues = OnsetUtils.onsetToAge(this.props.birthdate, prevState.onsetDate);
+        if (!!prevState.onsetDate || prevState.granularity !== granularity) {
```

Comment:
This was probably me while I was prototyping 🤔but this could perhaps now be moved to a separate method that can be invoked whenever values need to be reset.

Reply:
Changed in 1417f94